src/main/java/frc/trigon/robot/subsystems/turret/toohardturret/ToohardTurretIO.java

```diff
@@ -0,0 +1,26 @@
+package frc.trigon.robot.subsystems.turret.toohardturret;
+
+import com.ctre.phoenix6.controls.PositionVoltage;
+import com.ctre.phoenix6.hardware.TalonFX;
+import edu.wpi.first.math.MathUtil;
+import edu.wpi.first.math.geometry.Pose2d;
+import edu.wpi.first.math.geometry.Rotation2d;
+import frc.trigon.robot.subsystems.turret.TurretIO;
+import frc.trigon.robot.subsystems.turret.TurretInputsAutoLogged;
+
+public class ToohardTurretIO extends TurretIO {
+    private final TalonFX motor = ToohardTurretConstants.MOTOR;
+    private final PositionVoltage positionRequest = new PositionVoltage(0, 0, ToohardTurretConstants.FOC_ENABLED, 0, 0, false);
```

Comment:
Look at [MotionMagic](https://v6.docs.ctr-electronics.com/en/stable/docs/api-reference/device-specific/talonfx/closed-loop-requests.html#motion-magic) and implement it. (This is why you didn't use a profile in the turret subsystem, but the sim will need a profile to match the behavior of the real robot.)